demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php

```diff
@@ -2704,6 +2710,36 @@ private function copyFromBlueprint(array $data, string $blueprintId, Procedure $
         return $additionalDataEvent->getProcedure();
     }
 
+    /**
+     * Will copy legend file and reference it to the created procedure.
+     *
+     * @throws Exception
+     */
+    private function copyLegend($blueprintId, ProcedureInterface $newProcedure): void
+    {
+        try {
+            $blueprint = $this->getProcedureWithCertainty($blueprintId);
+            $legendStringFromBlueprint = $blueprint->getSettings()->getPlanPDF();
+
+            if ('' === $legendStringFromBlueprint) {
+                return;
+            }
+
+            // copy legend
+            $copiedFile = $this->fileService->copyByFileString($legendStringFromBlueprint, $newProcedure->getId());
+
+            // set planPDF with the referenced legends file and update procedure setting
+            $newProcedurePlanPdf = 'kopie_von'.$copiedFile->getName().':'.$copiedFile->getId().':'.$copiedFile->getMimetype();
```

Comment:
I would not call it 'kopie_von'. I would just use `$newProcedurePlanPdf = $copiedFile->getFileString()`

Reply:
done in https://github.com/demos-europe/demosplan-core/pull/3090/commits/5777f37e617b6f10ee5c2c5d71cbd81579cafeaf , thx